indico/modules/rb_new/client/js/common/rooms/RoomEditModal.jsx

```diff
@@ -0,0 +1,524 @@
+        } catch (error) {
+            handleAxiosError(error);
+            return;
+        }
+        this.setState({roomAvailability: camelizeKeys(response.data)});
+    }
+
+    async fetchRoomEquipment() {
+        const {roomId} = this.props;
+        let response;
+        try {
+            response = await indicoAxios.get(fetchRoomEquipmentURL({room_id: roomId}));
+        } catch (error) {
+            handleAxiosError(error);
+            return;
+        }
+        this.setState({roomEquipment: camelizeKeys(response.data)});
+    }
+
+    handleCloseModal = () => {
+        const {onClose} = this.props;
+        onClose();
+    };
+
+    handleSubmit = async (formData) => {
+        const {actions: {updateRoom}, room: {id}} = this.props;
+        const rv = await updateRoom(id, formData);
+        if (rv.error) {
+            return rv.error;
+        }
+    };
+
+    renderImage = (position) => {
+        return <SpriteImage pos={position} />;
+    };
+
+    removeAttribute = (attribute) => {
+        const {roomAttributes} = this.state;
+        this.setState({roomAttributes: _.omit(roomAttributes, attribute)});
+    };
+
+
+    renderAttributes = () => {
+        const {attributes, roomAttributes} = this.state;
+        if (!attributes) {
+            return;
+        }
+        return attributes.map(attribute => (
+            <>
+                {_.has(roomAttributes, attribute.name) && (
+                    <Field name={attribute.name}
+                           component={ReduxFormField}
+                           label={attribute.name}
+                           as={Input}
+                           isEqual={_.isEqual}
+                           icon={{name: 'trash', color: 'red', link: true, onClick: () => this.removeAttribute(attribute.name)}} />
+                )}
+            </>
+
+        ));
+    };
+
+    renderEquipmentList = ({input, ...props}) => {
+        return (
+            <ReduxFormField {...props}
+                            input={input}
+                            as={EquipmentList}
+                            componentLabel={Translate.string('Add new equipment')} />
+        );
+    };
+
+    renderPrincipalSearchField = (props) => {
+        const {favoriteUsers} = this.props;
+        return (
+            <ReduxFormField {...props}
+                            favoriteUsers={favoriteUsers}
+                            as={PrincipalSearchField}
+                            label={Translate.string('Owner')} />
+        );
+    };
+
+    renderColumn = (column, key) => (
+        <Grid.Column key={key}>{column.map(this.renderContent)}</Grid.Column>
+    );
+
+    renderContent = (content) => {
+        const {room, attributes, roomEquipment, roomAttributes, roomAvailability} = this.state;
+        const {equipmentTypes} = this.props;
+        if (!equipmentTypes) {
+            return;
+        }
+        let options = [];
```

Comment:
This is only used in one of the `case` blocks. Why not move it there?

Reply:
It was used in two case blocks before and it will probably be used in none. It's still wip.